spring-batch-docs/asciidoc/readersAndWriters.adoc

```diff
@@ -2779,221 +2779,255 @@ reconstituted from the execution context if the stream was
 
 [[decorators]]
 ==== Decorators
 
-In some cases a user needs a specialized behavior to be appended to a
-pre-existing ItemReader.   Spring Batch offers some out of the box decorators
-that can add additional behavior to to your `ItemReaders` and `ItemWriters`.
+In some cases, a user needs specialized behavior to be appended to a
+pre-existing `ItemReader`.   Spring Batch offers some out of the box decorators
+that can add additional behavior to to your `ItemReader` and `ItemWriter` implementations.
+
+Spring Batch includes the following decorators:
+
+* <<synchronizedItemStreamReader>>
+* <<singleItemPeekableItemReader>>
+* <<multiResourceItemWriter>>
 
 [[synchronizedItemStreamReader]]
-===== SynchronizedItemStreamReader
+===== `SynchronizedItemStreamReader`
 When using a `ItemReader` that is not thread safe, Spring Batch offers the
 `SynchronizedItemStreamReader` decorator that can be used to make the `ItemReader`
  thread safe. Spring Batch provides a
 `SynchronizedItemStreamReaderBuilder` to construct an instance of the
 `SynchronizedItemStreamReader`.
 
 [[singleItemPeekableItemReader]]
-===== SingleItemPeekableItemReader
-A decorator that adds a peek method to an `ItemReader`.
-This peek method allow the user to peek one item ahead.  Repeated calls to the
-peek will return the same item, and this will be the next item returned from read method.
+===== `SingleItemPeekableItemReader`
+Spring Batch includes a decorator that adds a peek method to an `ItemReader`.
+This peek method lets the user peek one item ahead.  Repeated calls to the
+peek returns the same item, and this is the next item returned from the `read` method.
 Spring Batch provides a `SingleItemPeekableItemReaderBuilder` to construct an
 instance of the `SingleItemPeekableItemReader`.
 
-NOTE: SingleItemPeekableItemReader's peek method is not thread-safe:
-it wouldn't be possible to honour the peek in multiple threads because only one
+NOTE: SingleItemPeekableItemReader's peek method is not thread-safe, because
+it would not be possible to honor the peek in multiple threads. Only one
 of the threads that peeked would get that item in the next call to read.
 
 [[multiResourceItemWriter]]
-===== MultiResourceItemWriter
+===== `MultiResourceItemWriter`
 The MultiResourceItemWriter wraps a `ResourceAwareItemWriterItemStream` and
-creates a new output resource when the count of items written in current
+creates a new output resource when the count of items written in the current
 resource exceeds the `itemCountLimitPerResource`.
 Spring Batch provides a `MultiResourceItemWriterBuilder` to construct an
 instance of the `MultiResourceItemWriter`.
 
 
 [[messagingReadersAndWriters]]
 ==== Messaging Readers And Writers
-Spring Batch offers readers and writers to some commonly used messaging systems.
+Spring Batch offers the following readers and writers for commonly used messaging systems:
+
+* <<AmqpItemReader>>
+* <<AmqpItemWriter>>
+* <<jmsItemReader>>
+* <<jmsItemWriter>>
```

Comment:
Are the jms supposed to be lower case here?

Reply:
They can be either lower or upper case (which I didn't realize). I would make them all lower case. That is, I would make the Amqp links read "amqp...." That way, they'll match the anchors to which they connect. If Asciidoc or Asciidoctor insists that the casing match in some future release, it won't go boom. Alternatively, you could make all the anchors lead with an upper-case letter.